content/basics/basics.mdx

```diff
@@ -0,0 +1,327 @@
+<AnimalsView animals={zoo.animals} />
+```
+
+`AnimalsView` automatically re-renders itself whenever you change the array, in
+this case by using `Zoo.addAnimal`. The individual `AnimalView` re-renders
+itself as before when any observed properties the underlying object change,
+which in this case is its `isHungry` status.
+
+## Local component state with `useLocalStore`
+
+So far we have discussed observable model state, where the models live outside
+of the React components. It is useful to have component-specific state however.
+This is state that only makes sense within that component, is typically very UI
+specific, and there is no reason to specify this part of your application's
+models.
+
+This is useful for "temporary data", such as for instance the text that a user
+is currently entering into an input widget. If the user then performs a special
+action such as pressing a button' this text may be saved into an actual model
+instance. And depending on your application's architecture you can even manage
+all your state this way.
+
+This is the MobX equivalent of component state as you'd use with the `useState`
+hook or the `setState` method in case of class components. Unlike React the
+`useState` hook, the MobX local store is observable.
+
+Here is an example where we add an animal to a zoo using a form:
+
+```tsx
+import { Component } from 'react'
+import { observer } from 'mobx-react'
+
+const AddAnimalView = observer(({ zoo }: { zoo: Zoo }) => {
+  const local = useLocalStore(() => ({
+    name: '',
+    species: '',
+  }))
+
+  const handleAdd = () => {
+    zoo.addAnimal(new Animal(local.name, local.species))
+    // clear input again after adding
+    local.name = ''
+    local.species = ''
+  }
+
+  const handleChangeName = (e: React.ChangeEvent<HTMLInputElement>) => {
+    local.name = e.target.value
+  }
+
+  const handleChangeSpecies = (e: React.ChangeEvent<HTMLInputElement>) => {
+    local.species = e.target.value
+  }
+
+  return (
+    <div>
+      <div>
+        Name:
+        <input type="text" value={local.name} onChange={handleChangeName} />
+      </div>
+      <div>
+        Species:
+        <input
+          type="text"
+          value={local.species}
+          onChange={handleChangeSpecies}
+        />
+      </div>
+      <button onClick={handleAdd}>Add animal</button>
+    </div>
+  )
+})
+```
+
+Here we define an object called `local` with two observable properties, `name`
+and `species`. We use the React [controlled
+components](https://reactjs.org/docs/forms.html#controlled-components)
+technique to allow the user to edit these properties. As you can see, this
+simply involves mutating the local state directly. We also have an `Add animal`
+button which uses the local state to modify the zoo using the `addAnimal`
+action. It then needs to reset the local state again to allow the user to
+enter another animal.
+
+XXX interaction with strict mode, see https://github.com/mobxjs/mobx-react-lite/issues/235
```

Comment:
XXX? :)

Reply:
Yes, a "TODO" comment. The problem is that this whole example breaks down, if I understand it correctly, as soon as you use MobX strict mode. On this issue you suggest useLocalStore is not worth it in strict mode. But what then would be the equivalent for the local component state use case? 

I want to write a document with *one* way to do it, and I either have to go to add text that says you shouldn't use MobX strict mode, or text that says, use MobX strict mode and this is what you use (but what? not useLocalStore).

Reply:
In addition I just read that MobX 6 wants to make 'observed' be the default. So if that doesn't go together with `useLocalStore` we need to go back to the drawing board completely for this section.

Reply:
Yes, you are right. I think we will need to rethink `useLocalStore` anyway with MobX 6, it won't be that useful anymore. I will open a new issue to track that.

Reply:
Actually, we are already tracking this problem in https://github.com/mobxjs/mobx-react-lite/issues/259. It would be probably best to remove `useLocalStore` from this document completely because it's uncertain how it will look like in the end.